examples/3d/parallax_mapping.rs

```diff
@@ -0,0 +1,356 @@
+        ..default()
+    });
+    commands.spawn((
+        PbrBundle {
+            mesh: meshes.add(cube),
+            material: parallax_material.clone_weak(),
+            ..default()
+        },
+        Spin { speed: 0.3 },
+    ));
+
+    let mut background_cube: Mesh = shape::Cube { size: 40.0 }.into();
+    background_cube.generate_tangents().unwrap();
+    let background_cube = meshes.add(background_cube);
+
+    let background_cube_bundle = |translation| {
+        (
+            PbrBundle {
+                transform: Transform::from_translation(translation),
+                mesh: background_cube.clone(),
+                material: parallax_material.clone(),
+                ..default()
+            },
+            Spin { speed: -0.1 },
+        )
+    };
+    commands.spawn(background_cube_bundle(Vec3::new(45., 0., 0.)));
+    commands.spawn(background_cube_bundle(Vec3::new(-45., 0., 0.)));
+    commands.spawn(background_cube_bundle(Vec3::new(0., 0., 45.)));
+    commands.spawn(background_cube_bundle(Vec3::new(0., 0., -45.)));
+
+    let style = TextStyle {
+        font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+        font_size: 18.0,
+        color: Color::WHITE,
+    };
+
+    commands.spawn(
+        TextBundle::from_sections(vec![
+            TextSection::new(
+                format!("Parallax depth scale: {parallax_depth_scale:.5}\n"),
+                style.clone(),
+            ),
+            TextSection::new(
+                format!("Layers: {max_parallax_layer_count:.0}\n"),
+                style.clone(),
+            ),
+            TextSection::new("Method: Relief Mapping\n", style.clone()),
+            TextSection::new("\n\n", style.clone()),
+            TextSection::new("Controls\n", style.clone()),
+            TextSection::new("---------------\n", style.clone()),
+            TextSection::new("Left click - Change view angle\n", style.clone()),
+            TextSection::new(
+                "1/2 - Decrease/Increase parallax depth scale\n",
+                style.clone(),
+            ),
+            TextSection::new("3/4 - Decrease/Increase layer count\n", style.clone()),
+            TextSection::new("Space - Switch parallaxing algorithm\n", style),
+        ])
+        .with_style(Style {
+            position_type: PositionType::Absolute,
+            top: Val::Px(10.0),
+            left: Val::Px(10.0),
+            ..default()
+        }),
+    );
+}
+
+/// Store handle of the normal to later modify its format in [`update_normal`].
+#[derive(Resource)]
+struct Normal(Option<Handle<Image>>);
+
+/// Work around the fact that the default bevy image loader sets the
+/// normal's format to something incompatible with normal shaders.
+/// The format must be one of the `TextureFormat` ending in `*Unorm`.
+///
+/// In this function, we wait until the image is loaded, immediately
+/// change its format and never run the core logic afterward.
+///
+/// Without proper format, lighting looks wrong.
```

Comment:
What format does the loader set?

Reply:
Nevermind. The culprit is `impl AssetLoader for ImageTextureLoader`. (`crates/bevy_render/src/texture/image_texture_loader.rs`).

When I use `asset_server.load("foobar.png")`, `is_srgb` will always be set to `true` when calling `Image::from_buffer` in `<ImageTextureLoader as AssetLoader>::load`. The resulting loaded asset is of format `Rgba8UnormSrgb`, which is invalid for normals afaik.

I can't influence that by modifying the asset, it's just hardcoded in the `load` code.

Reply:
Ah, I see. Just an sRGB issue. Could you update the comment to point out that the depth map texture is linear but the image loader does not yet know whether an arbitrary image asset should be sRGB or not, and it defaults to sRGB because most of the time people are loading images outside of models to use as colour textures?

Reply:
Are you commenting on the latest version? I've already updated the comment to explain more why I overwrite the format and specifically point out that it's because it's a normal map and you shouldn't do it generally.

There is no worries to have with regard to the depth map, as far as I can tell, it works as expected.

In details, I'm actually very confused on this. I'm not sure if overwriting the format is correct here. It seems so? Because otherwise the error would be clearly visible. But playing around with exporting options from GIMP, I still can't tell if I'm using a linear color space. Afaik, it's still sRGB, but as mentioned, it works, so it should be the correct format.

Reply:
Ahhhh sorry. I must have followed a link from an email which takes me to code at that version. I was reviewing old code.